reana_workflow_controller/rest/utils.py

```diff
@@ -25,38 +25,66 @@
 from reana_commons.utils import get_workflow_status_change_verb
 from reana_db.database import Session
 from reana_db.models import Job, JobCache, Workflow, WorkflowStatus
+from sqlalchemy.exc import SQLAlchemyError
+from kubernetes.client.rest import ApiException
 
 from reana_workflow_controller.config import (REANA_GITLAB_HOST,
                                               WORKFLOW_TIME_FORMAT)
-from reana_workflow_controller.errors import (REANAWorkflowControllerError,
+from reana_workflow_controller.errors import (REANAExternalCallError,
+                                              REANAWorkflowControllerError,
                                               REANAWorkflowDeletionError)
 from reana_workflow_controller.workflow_run_manager import \
     KubernetesWorkflowRunManager
 
 
 def start_workflow(workflow, parameters):
     """Start a workflow."""
-    if workflow.status in [WorkflowStatus.created, WorkflowStatus.queued]:
+    current_db_sessions = Session.object_session(workflow)
+    kwrm = KubernetesWorkflowRunManager(workflow)
+
+    def _start_workflow_db(workflow, parameters):
```

Comment:
For the sake of readability, what do you think about moving these private functions to the top of `start_workflow`?